### ISSUE TITLE
Intellisense and TSX: not working when "foreign" namespace

### ISSUE THREAD

**Issue by TheWind3** [Microsoft/TypeScript#7388]
Visual Studio: No intellisense in TSX file in certain conditions.
However, compilation is OK.

**TypeScript Version:** 
1.8.6

**VisualStudio Version:** 
2015, 14.0.24720.00 update 1

**Code**
```ts
// filename: test.tsx
declare module JSX {
  interface Element {
  }
  interface ElementAttributesProperty {
    props;
  }
  interface IntrinsicElements {
    div: any;
  }
}
namespace Library {
  export class LibComponent {
    props: {
      title: string;
    };
  }
}
namespace Application {
  import lib = Library;

  class AppComponent {
    props: {
      title: string;
    };
  }

  function appMain(): void {
    var result = (
      <div>
// get intellisense here:
        <AppComponent title="My Application" />
        <AppCompone // get intellisense
        <AppComponent tit // get intellisense

// no intellisense here:
        <Library.LibComponent title="Library item 1" />
        <Libra // no intellisense
        <Library. // no intellisense
        <Library.LibCo // no intellisense
        <Library.LibComponent tit // no intellisense

// no intellisense here:
        <lib.LibComponent title="Library item 1" />
        <li // no intellisense
        <lib. // no intellisense
        <lib.LibCo // no intellisense
        <lib.LibComponent tit // no intellisense
      </div>
    );
  }
}

```

**Comment by RyanCavanaugh** [comment]
Should be fixed by #7326 (see #4577)

**Comment by RyanCavanaugh** [comment]
Verified
